### PR TITLE
Add recipe for ELSI interface

### PR DESCRIPTION
**Pull request by awvwgk**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by awvwgk** [comment]
@conda-forge/staged-recipes Please have a look at this recipe for a Fortran/MPI interface library, please let me know if there is anything that needs to addressed before this can be merged.

Thanks in advance.

File: recipes/elsi/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    - liblapack
+  run:
+    - {{ mpi }}
+    - libomm * {{ mpi_prefix }}_*
+    - matrixswitch * {{ mpi_prefix }}_*
+    - elpa * {{ mpi_prefix }}_*
+    - ntpoly * {{ mpi_prefix }}_*
+    - scalapack
+    - libblas
+    - liblapack
```
Comment (beckermr) [inline recipes/elsi/meta.yaml:50]: You shouldn't list these in run.

File: recipes/elsi/conda_build_config.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+mpi:
+  - mpich
+  - openmpi
+
+pin_run_as_build:
+  mpich: x.x
+  openmpi: x.x
```
Comment (beckermr) [inline recipes/elsi/conda_build_config.yaml:7]: You shouldn't need these keys either.

File: recipes/elsi/meta.yaml (hunk 1 above)
Comment (awvwgk) [reply recipes/elsi/meta.yaml:50]:
```suggestion
```

File: recipes/elsi/conda_build_config.yaml (hunk 2 above)
Comment (awvwgk) [reply recipes/elsi/conda_build_config.yaml:7]:
```suggestion
```

File: recipes/elsi/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+about:
+  home: https://wordpress.elsi-interchange.org/
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file:
+    - LICENSE
+    # FortJSON is BSD-3-Clause, but developed by the ELSI team
+    # https://gitlab.com/elsi_project/elsi_interface/-/issues/52
+    - external/FortJSON/LICENSE
```
Comment (beckermr) [inline recipes/elsi/meta.yaml:68]: I can't tell if anything is still vendored or not. The licenses of any vendored libraries need to go here. Feel free to copy the license files into the recipe if needed. The comments on the PR indicate that one library is still vendored?

Comment (beckermr) [reply recipes/elsi/meta.yaml:68]: If stuff is still vendored, you can use SPDX expressions in the license field (e.g, `BSD-3-Clause AND ...`).

Comment (awvwgk) [reply recipes/elsi/meta.yaml:68]: FortJSON is still vendored, but cannot be devendored without changes in the upstream project at the moment. I opened an issue upstream for this. Everything else is properly included from dependencies on conda-forge.

Comment (awvwgk) [reply recipes/elsi/meta.yaml:68]: The only vendored dependency is FortJSON which is also BSD-3-Clause. I just looked into patching the build files to devendor it myself, but FortJSON has no release which could be packaged. Opened an issue upstream for this as well (https://gitlab.com/elsi_project/fortjson/-/issues/2).

**Comment by beckermr** [comment]
Sorry for the delay on this one. Thank you!

**Comment by awvwgk** [comment]
Thanks for reviewing!

**Pull request merged**